src/Components/Blazor/Build/src/Tasks/GenerateBlazorBootJson.cs

```diff
@@ -72,15 +119,43 @@ public class BootJsonData
             public string entryAssembly { get; set; }
 
             /// <summary>
-            /// Gets the closure of assemblies to be loaded by Blazor WASM. This includes the application entry assembly.
+            /// Gets the set of resources needed to boot the application. This includes the transitive
+            /// closure of .NET assemblies (including the entrypoint assembly), the dotnet.wasm file,
+            /// and any PDBs to be loaded.
             /// </summary>
-            public string[] assemblies { get; set; }
+            public Dictionary<ResourceType, ResourceList> resources { get; set; }
+
+            /// <summary>
+            /// Gets a value that determines whether to enable caching of the <see cref="resources"/>
+            /// inside a CacheStorage instance within the browser.
+            /// </summary>
+            public bool cacheBootResources { get; set; }
+
+            /// <summary>
+            /// Gets a value that determines if this is a debug build.
+            /// </summary>
+            public bool debugBuild { get; set; }
 
             /// <summary>
             /// Gets a value that determines if the linker is enabled.
             /// </summary>
             public bool linkerEnabled { get; set; }
         }
+
+        public enum ResourceType
+        {
+            assembly,
+            pdb,
+            wasm
+        }
+
+        /// <summary>
+        /// Represents a set of resources used when booting a Blazor WebAssembly application.
+        /// The dictionary keys are the resource names; values are SHA-256 hashes of the corresponding files.
+        /// </summary>
+        public class ResourceList : Dictionary<string, string>
```

Comment:
`using ResourceList = Dictionary<string, string>`? Also this is clearly not a list.

Reply:
Done